### ISSUE TITLE
metadata error in output of `prepare-data convert-citations`

### ISSUE THREAD

**Issue by senderle** [agoldst/dfr-browser#5]
I'm working on adapting the topic modeling tool to work with this wonderful piece of software. In the process of getting a testing workflow set up, I found that the standard `citations.tsv` file provided by the DFR tool, when passed to `prepare-data`, appears to generate a rogue line like this at the end:

```
"10.2307/461391","The Telltale Teeth: [...]
"",""                                         < -- here lieth a TypeError
```

JavaScript becomes confused when it tries to read the data from this line, and dies.

I think this is caused by an unfortunate blank line at the end of `citations.tsv`. Fixing should be as simple as removing any blank lines from the input before processing, but I wanted to pass it by you before submitting a PR.

**Comment by agoldst** [comment]
Thanks, Scott, for catching this. I have been doing all my data-converting using my R package and it's been a long while since I tested these python scripts carefully. Yes, by all means send a PR to fix prepare-data; that would be most welcome.

**Comment by agoldst** [comment]
PS would be delighted to if you're able to make use of this in your own project!

**Comment by senderle** [comment]
[Done](https://github.com/agoldst/dfr-browser/pull/6).

My plan, right now, is to wipe out the bad old HTML output that the TMT generates in favor of your browser, which is much more pleasant to look at, and more informative to boot. If I am successful, I think that many, many projects will be able to make use of this!

The current tool generates static pages, which can get _huge_. I assume that your dynamic loading strategy is much better overall. By the time dynamic loading method starts to reach scaling limits, I suspect the static approach would also become impractical.

 I also want to try adding a javascript-based pivot table ([eg.](https://github.com/nicolaskruchten/pivottable)) to the output. I'm not certain it will work well; we'll see.

**Comment by agoldst** [comment]
Fixed by #6. Thanks so much for the patch. What I had failed to notice was that I had a bunch of local commits on master that I hadn't pushed up, so to keep myself sane I have pushed those up too. They shouldn't change any functionality and hopefully they don't break anything. They are preparatory to a bigger rewrite that allows for swapping between models / metadata covariates in a browser, which I finished months ago but have not had time to test or update the data-prep scripts for. Sigh.
